### PR TITLE
Update contributing

### PR DESCRIPTION
**Pull request by Michaelangel007**
@tomcw Here is a draft of the updated contributing doc.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,20 +24,44 @@ Before submitting an enhancement, search the existing issues to see if this enha
...
+
+When submitting UI changes please discuss WHY you are making changes.
+
+* **Our userbase expects things to work they way they are used to.**  Changing default behavior without a "migration path" causes users to submit bug reports asking us to revert changes which wastes everyone's time.
+
+* **Functionality should be prioritized over Form.** The _entire_ point of UI is to empower the user to do the thing they want to do and _then get out of their way._  A pretty UI, but one that frustrates users, is not empowering them.
+
+* Also keep in mind many so called modern UI/UX "experts" tend to make changes for the sake of change in order to sell a product. Worse they tend to focus on mobile design even when it makes little or no sense on a desktop platform. There is a time and a place to focus on Form but as a mature project we are more interested in stability and functionality then throwing on a fresh coat of paint that more likely to introduces a new set of bugs.
```
Comment (tomcw) [inline CONTRIBUTING.md:57]: typo: "then" -> "than"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,20 +24,44 @@ Before submitting an enhancement, search the existing issues to see if this enha
...
+
+Understand that not everyone will agree to UI changes.  The developers have been using computers a very long time and are not interested in chasing modern UI fads unless it a good reason can be shown _why_ the UI should be changed.  For example, take a poll, or even better raise a issue asking for feedback.  This way we have hard data showing that there is interest instead of assuming that a change is automatically "better".
+
+Trivial changes such as updating art are much easier to accept then rewriting the UI moving buttons around.
```
Comment (tomcw) [inline CONTRIBUTING.md:61]: typo: "then" -> "than"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -24,20 +24,44 @@ Before submitting an enhancement, search the existing issues to see if this enha
...
 
 For new (large) features, then link the PR back to an enhancement issue, where the proposed feature had been discussed with AppleWin maintainers first and justified using suitable use-cases. In this enhancement issue be sure to include a specification of the feature, and a design if necessary. Having a design (doc, diagrams) that explains the logic/algorithms/protocol etc will help in the long term maintenance of this feature.
 
 When changing project metadata files (eg. .sln, .vcproj, .rc) then different versions of Visual Studio may decide to reorder or re-format other sections of the file. This can result in lots of churn in the file each time a trivial edit is made. So before committing the PR, check the difference, and if there's been lots of unnecessary changes then just make the required change using a non-Visual Studio editor.
+
+When submitting UI changes please discuss WHY you are making changes.
+
+* **Our userbase expects things to work they way they are used to.**  Changing default behavior without a "migration path" causes users to submit bug reports asking us to revert changes which wastes everyone's time.
```
Comment (tomcw) [inline CONTRIBUTING.md:53]: typo: "they way" -> "the way"

**Comment by tomcw** [comment]
I've found a few typos, but otherwise it looks fine.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Michaelangel007) [reply CONTRIBUTING.md]: Bloody hell.  One of these days I'll get `then` and `than` [sorted out.](https://theoatmeal.com/comics/misspelling)

Anyhoo, thanks for catching these Tom!

**Comment by Michaelangel007** [comment]
@tomcw Should all be good now.  I'll leave the conversations open until you've manually confirm all the fixes are in.

**Comment by sicklittlemonkey** [comment]
If I'm seeing the latest version, I think these corrections apply:
"Next, please make sure your code **mades**" -> "**matches**"?
"fresh coat of paint that more likely to introduces a new set of bugs" -> reword or remove "**to**".
"not interested in chasing modern UI fads unless it a good reason can be shown" -> reword or remove "**it**".

**Comment by Michaelangel007** [comment]
@sicklittlemonkey  Thanks for the catches Nick!  Fixed.

**Comment by tomcw** [comment]
LGTM. So go ahead and "Squash & merge".

**Pull request merged**